src/environment/shell_unix.go

```diff
@@ -102,7 +102,7 @@ func (env *ShellEnvironment) InWSLSharedDrive() bool {
 		return false
 	}
 	windowsPath := env.ConvertToWindowsPath(env.Pwd())
-	return !strings.HasPrefix(windowsPath, `\\wsl.localhost`)
+	return !strings.HasPrefix(windowsPath, `//wsl.localhost`) && !strings.HasPrefix(windowsPath, `//wsl$`)
```

Comment:
BTW, I don't perform only one check on a `//wsl` prefix here because a UNC path to a non-WSL filesystem can have this prefix as well.

Reply:
gotcha